src/services/completions.ts

```diff
@@ -223,6 +227,20 @@ namespace ts.Completions {
         return uniques;
     }
 
+    function isInBreakOrContinue(sourceFile: SourceFile, position: number): boolean {
+        const contextToken = findPrecedingToken(position, sourceFile);
+        return contextToken && isBreakOrContinueStatement(contextToken.parent) &&
+            (contextToken.kind === SyntaxKind.BreakKeyword || contextToken.kind === SyntaxKind.ContinueKeyword || contextToken.kind === SyntaxKind.Identifier);
+    }
+
+    function getLabelCompletionAtPosition(sourceFile: SourceFile, position: number): CompletionInfo | undefined {
+        const contextToken = findPrecedingToken(position, sourceFile);

```

Comment:
This was just computed in `isBreakOrContinue`, move this outside so it's only computed once.